examples/piecestore/rpc/client/main.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/urfave/cli"
+	"github.com/zeebo/errs"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/examples/piecestore/rpc/client/api"
+	"storj.io/storj/examples/piecestore/rpc/client/utils"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+var argError = errs.Class("argError")
+
+func main() {
+
+	app := cli.NewApp()
+
+	// Set up connection with rpc server
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(":7777", grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %s", err)
+	}
+	defer conn.Close()
+	client := pb.NewPieceStoreRoutesClient(conn)
+	ctx := context.Background()
+
+	app.Commands = []cli.Command{
+		{
+			Name:    "upload",
+			Aliases: []string{"u"},
+			Usage:   "upload data",
+			Action: func(c *cli.Context) error {
+				if c.Args().Get(0) == "" {
+					return argError.New("No input file specified")
+				}
+
+				file, err := os.Open(c.Args().Get(0))
+				if err != nil {
+					return err
+				}
+				// Close the file when we are done
+				defer file.Close()
+
+				fileInfo, err := os.Stat(c.Args().Get(0))
+
+				if fileInfo.IsDir() {
+					return argError.New(fmt.Sprintf("Path (%s) is a directory, not a file", c.Args().Get(0)))
+				}
+
+				var fileOffset, storeOffset int64 = 0, 0
+				var length = fileInfo.Size()
+				var ttl = time.Now().Unix() + 86400
+
+				id, err := utils.DetermineID(file, fileOffset, length)
+				if err != nil {
+					return err
+				}
+
+				// Created a section reader so that we can concurrently retrieve the same file.
+				dataSection := io.NewSectionReader(file, fileOffset, length)
+
+				err = api.StorePieceRequest(ctx, client, id, dataSection, fileOffset, length, ttl, storeOffset)
+
+				if err != nil {
+					fmt.Printf("Failed to store file of id: %s\n", id)
+					return err
+				}
+
+				fmt.Printf("successfully Stored file of id: %s\n", id)
+				return nil
+			},
+		},
+		{
+			Name:    "download",
+			Aliases: []string{"d"},
+			Usage:   "download data",
+			Action: func(c *cli.Context) error {
+				if c.Args().Get(0) == "" {
+					return argError.New("No id specified")
+				}
+
+				id := c.Args().Get(0)
+
+				if c.Args().Get(1) == "" {
+					return argError.New("No output file specified")
+				}
+				_, err := os.Stat(c.Args().Get(1))
+
+				if !os.IsNotExist(err) {
+					return argError.New("Path already exists: " + c.Args().Get(1))
+				}
```

Comment:
another comment i've left before -  what happens if the err is not nil, but the err is also not a NotExist error?

gotta still check that the err != nil in general

Reply:
also from a formatting perspective i'd prefer there be a newline before the os.Stat and not after

Reply:
👍